### ISSUE TITLE
Accented chars in buildFilterRegex string conversion

### ISSUE THREAD

**Issue by ethieblin** [sparna-git/Sparnatural#471]
Hello,

I am using a SPARQL endpoint with language in French (therefore it contains non-ascii chars). I am using a Virtuoso.
I have a property "title" for the sparnatural configuration which inherits from the sparnatural-config-core:SearchProperty.

When a title is searched for in the interface with the word "qualité" for example, the generated query looks like:

```
SELECT DISTINCT ?Thing_1 ?Title_2 WHERE {
  ...
  ?Thing_1 dc:title ?Title_2.
  FILTER(REGEX(STR(?Title_2), "qualité", "i"))
}
```
Which gives no result, whereas the same query without the string casting works fine:
```
SELECT DISTINCT ?Thing_1 ?Title_2 WHERE {
  ...
  ?Thing_1 dc:title ?Title_2.
  FILTER(REGEX(?Title_2, "qualité", "i"))
}
```

If you want to try on some open data:
[data.bnf query with the STR](https://data.bnf.fr/sparql/#query=PREFIX%20rdfs%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2F2000%2F01%2Frdf-schema%23%3E%0ASELECT%20DISTINCT%20*%0AWHERE%20%7B%0A%20%20%3Chttp%3A%2F%2Fdata.bnf.fr%2Ftemp-work%2F0051aa7d1faf962ea3c31e9901e3a14d%2F%23about%3E%20rdfs%3Alabel%20%3Fy.%0A%20%20FILTER(REGEX(STR(%3Fy)%2C%20%22qualit%C3%A9%22%2C%20%22i%22))%0A%7D%20LIMIT%20100&endpoint=https%3A%2F%2Fdata.bnf.fr%2Fsparql&requestMethod=POST&tabTitle=Query&headers=%7B%7D&contentTypeConstruct=text%2Fturtle%2C*%2F*%3Bq%3D0.9&contentTypeSelect=application%2Fsparql-results%2Bjson%2C*%2F*%3Bq%3D0.9&outputFormat=table)
[data.bnf query without the STR](https://data.bnf.fr/sparql/#query=PREFIX%20rdfs%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2F2000%2F01%2Frdf-schema%23%3E%0ASELECT%20DISTINCT%20*%0AWHERE%20%7B%0A%20%20%3Chttp%3A%2F%2Fdata.bnf.fr%2Ftemp-work%2F0051aa7d1faf962ea3c31e9901e3a14d%2F%23about%3E%20rdfs%3Alabel%20%3Fy.%0A%20%20FILTER(REGEX(%3Fy%2C%20%22qualit%C3%A9%22%2C%20%22i%22))%0A%7D%20LIMIT%20100&endpoint=https%3A%2F%2Fdata.bnf.fr%2Fsparql&requestMethod=POST&tabTitle=Query&headers=%7B%7D&contentTypeConstruct=text%2Fturtle%2C*%2F*%3Bq%3D0.9&contentTypeSelect=application%2Fsparql-results%2Bjson%2C*%2F*%3Bq%3D0.9&outputFormat=table)

Would it be possible to remove this "STR" casting in buildFilterRegex ? or add something in the configuration to skip it ?
Thanks :)

**Comment by tfrancart** [comment]
> Would it be possible to remove this "STR" casting in buildFilterRegex ?

No, it is here on purpose. Because we want REGEX to work on any kind of literals, not only strings, and because we have the opposite behavior when using GraphDB : without STR it will not work.
This is inline with the SPARQL Spec : https://www.w3.org/TR/2013/REC-sparql11-query-20130321/#restrictString _"SPARQL FILTER functions like [regex](https://www.w3.org/TR/2013/REC-sparql11-query-20130321/#func-regex) can test RDF literals. regex matches only [string literals](https://www.w3.org/TR/2013/REC-sparql11-query-20130321/#func-string). regex can be used to match the lexical forms of other literals by using the [str](https://www.w3.org/TR/2013/REC-sparql11-query-20130321/#func-str) function."_

> or add something in the configuration to skip it ?

It cannot be done easily and I am relunctant to do that since this looks like a high price to pay to workaround what looks like a Virtuoso bug. Let's keep this open and we'll see in the future.

**Comment by ethieblin** [comment]
Ok, thank you for your answer
